### ISSUE TITLE
one by one means inputing 8 test samples every iteration?

### ISSUE THREAD

**Issue by XYHhhhhhhhh** [sylyoung/DeepTransferEEG#1]
In the article, it was mentioned that "The target EEG trials arrive online one by one, so we perform incremental EA on them.",  however in the algorithms part you said "SFUDA and TTA used the same learning rate and optimizer, but test batch size 8." So one by one means inputing 8 test samples every iteration?

**Comment by sylyoung** [comment]
> In the article, it was mentioned that "The target EEG trials arrive online one by one, so we perform incremental EA on them.", however in the algorithms part you said "SFUDA and TTA used the same learning rate and optimizer, but test batch size 8." So one by one means inputing 8 test samples every iteration?


SFUDA conducts offline training, with non-overlapping test batch size 8.
TTA conducts online training at test-time, using test batch size 8 with [a-8, a) samples at test time a, i.e., past 8 samples containing the currect test sample. # zero indexed

As an example, suppose the test subject has 40 trials/samples.
SFUDA loops through 5 batches, i.e., [0, 8), [8, 16), [16, 24), [24, 32) , [32, 40) for some epochs, the model is updated (5 x epoch_num) steps in total. 
TTA uses sliding batchs, i.e., [0, 8), [1, 9), [2, 10), [3, 11), ..., [32, 40) for a total of 33 batches. There are no ”loops“ or "epochs" in this case. The model is updated 33 steps in total, one step per batch. Note that this is the very last step of our  algorithm. For the 11-th sample (index 10 in zero-indexing), the order of the algorithm is EA on [0, 11) -> predict on 10 -> update using [3, 11) batch
